src/chains.rs

```diff
@@ -537,17 +542,27 @@ impl Rewrite for Chain {
 
         formatter.format_root(&self.parent, context, shape)?;
         if let Some(result) = formatter.pure_root() {
-            return wrap_str(result, context.config.max_width(), shape);
+            return wrap_str(result, context.config.max_width(), shape)
+                .max_width_error(shape.width, self.parent.span);
         }
 
+        let first = self.children.first().unwrap_or(&self.parent);
+        let last = self.children.last().unwrap_or(&self.parent);
+        let children_span = mk_sp(first.span.lo(), last.span.hi());
+        let full_span = self.parent.span.with_hi(children_span.hi());
+
         // Decide how to layout the rest of the chain.
-        let child_shape = formatter.child_shape(context, shape)?;
+        let child_shape = formatter
+            .child_shape(context, shape)
+            .max_width_error(shape.width, children_span)?;
 
         formatter.format_children(context, child_shape)?;
         formatter.format_last_child(context, shape, child_shape)?;
 
-        let result = formatter.join_rewrites(context, child_shape)?;
-        wrap_str(result, context.config.max_width(), shape)
+        let result = formatter
+            .join_rewrites(context, child_shape)
+            .max_width_error(child_shape.width, children_span)?;
```

Comment:
Since `pure_root` returns `Some` if the chain is only a root, `None` if the chain has its children, I'm not sure whether there is any error to propagate up. However, in case of `join_rewrites`, I think we can make that change after making one small decision

Reply:
Got it. If that's the case then maybe it makes more sense to leave this as `Option<Sting>` for now.